source/agora/node/Validator.d

```diff
@@ -411,4 +415,29 @@ public class Validator : FullNode, API
         super.shutdown();
         this.nominator.storeLatestState();
     }
+
+    /***************************************************************************
+
+        Check if the current enrollment is about to expire and re-enroll
+        if necessary.
+
+        Params:
+            block_height = Current block height
+
+    ***************************************************************************/
+
+    private void checkEnrollmentExpiration (Height block_height) @safe
```

Comment:
The function should be renamed. It does more than just check, it also re-enrolls.

I'm not sure what a better name would be without making it very long.

Reply:
Maybe `checkAndReEnroll`? Looks a bit ugly though.

Reply:
`renewExpiringEnrollment` ? It is equally ugly tho :)

Reply:
`checkAndEnroll` could work. I mean the fact that it's a "re"-enrollment doesn't really have that much of a semantical difference, my two cents..